packages/btcindexer/src/api/client.ts

```diff
@@ -16,17 +16,27 @@ const msgPackHeaders = {
 
 export default class Client {
 	baseUrl: string;
+	bearerToken?: string;
 
-	constructor(baseUrl: string) {
+	constructor(baseUrl: string, bearerToken?: string) {
 		if (baseUrl.endsWith("/")) baseUrl = baseUrl.slice(0, -1);
 
 		this.baseUrl = baseUrl;
+		this.bearerToken = bearerToken;
+	}
+
+	private getHeaders(additionalHeaders: Record<string, string> = {}): Record<string, string> {
```

Comment:
This function is not simply extracting headers. It's construction them, so let's rename.

```suggestion
    // makes headers and injects bearer authentication header.
	private mkHeaders(additionalHeaders: Record<string, string> = {}): Record<string, string> {
```

Reply:
done